src/python/grpcio/support.py

```diff
@@ -76,16 +76,21 @@ def _expect_compile(compiler, source_string, error_message):
         "Diagnostics found a compilation environment issue:\n{}"
             .format(error_message))
 
-def diagnose_build_ext_error(build_ext, error):
-  {
-      errors.CompileError: diagnose_compile_error
-  }[type(error)](build_ext, error)
-
 def diagnose_compile_error(build_ext, error):
   """Attempt to run a few test files through the compiler to see if we can
      diagnose the reason for the compile failure."""
   for c_check, message in C_CHECKS.items():
     _expect_compile(build_ext.compiler, c_check, message)
+
+ERROR_DIAGNOSES = {
+    errors.CompileError: diagnose_compile_error
+}
+
+def diagnose_build_ext_error(build_ext, error, formatted):
+  if error in ERROR_DIAGNOSES:
```

Comment:
Should this not be `type(error)` on this line?

What do you think of the style of only looking into the dictionary once:
```
diagnosis = _ERROR_DIAGNOSES.get(type(error))
if diagnosis is None:
  raise commands.CommanddError(<...>)
else:
  diagnosis(build_ext, error)
```
?